### ISSUE TITLE
Wsuggest-override with AFQMC

### ISSUE THREAD

**Issue by ye-luo** [QMCPACK/qmcpack#3391]
**Describe the bug**
There are override issues left in AFQMC. We need to address and then turn on AFQMC in github action Werror builds.
This may also need fix in Multi.

**To Reproduce**
Add -DBUILD_AFQMC=ON. With -DENABLE_CUDA=ON may show more warning.
https://jenkins-ci.cels.anl.gov/job/QMCPACK-CI-Clang13-AFQMC-Offload-CUDA/81/label=compute002.mcs.anl.gov,type=Real-enable-CUDA/consoleFull

**Expected behavior**
No warning by GCC.

**Comment by ye-luo** [comment]
@williamfgc @quantumsteve could you have a look at this? If changes are needed in external_codes source code like mpi_wrapper and boost_multi, contact @correaa . If warnings are from mpi header files for example openmpi, we just leave them.

**Comment by quantumsteve** [comment]
@correaa I see you already [added the override keyword to this line in boost-multi](https://gitlab.com/correaa/boost-multi/-/blob/2e4d5f4e59ef346ee8deedf9f9924023c374ecf8/memory/monotonic.hpp#L101). How can we bring this change into qmcpack?

**Comment by correaa** [comment]
> @correaa I see you already [added the override keyword to this line in boost-multi](https://gitlab.com/correaa/boost-multi/-/blob/2e4d5f4e59ef346ee8deedf9f9924023c374ecf8/memory/monotonic.hpp#L101). How can we bring this change into qmcpack?

There is a protocol that @markdewing  put in place. 
Since it is using subtree feature of git it goes something like this:

```
git rm -r external_codes/boost_multi/multi
git commit -m "remove multi subtree"
git subtree add --squash -P external_codes/boost_multi/multi $CI_REPOSITORY_URL $CI_COMMIT_BRANCH # e.g. https://gitlab.com/correaa/boost-multi.git
```
I don't know of a better way.

But first I have to merge this https://github.com/correaa/qmcpack/pull/1. 
I just merged it locally, but I don't know how to bring that pull request into qmcpack main repository.
I did it online and for some reason it is not creating a PR in qmcpack. Do you know how to do that?

**Comment by quantumsteve** [comment]
You need to change the base repository from correa/qmcpack to QMCPACK/qmcpack

I think the easiest way is create the PR from the link below
https://github.com/QMCPACK/qmcpack/compare/develop...correaa:ppconvert_use_iostream

**Comment by correaa** [comment]
Thank you very much, that is exactly what I needed.
Yes, I looked for the option of the base repository but didn't see qmcpack and I don't remember needing to do that because official qmcpack is still the upstream (?) repository. Maybe what is different this time is that I did it all with the github interface.

Ok, so once https://github.com/QMCPACK/qmcpack/pull/3481 is merged, I can make a small change in multi, merge into multi master and the subtree update can be done in qmcpack.

If you know of a better workflow let me know.
